### DISCUSSION TITLE
single pole figure for all Miller Indices

### DISCUSSION THREAD

**Discussion by amirtopaz** [mtex-toolbox/mtex#724] [Ask Anything] [answered]
Hi,
I am trying to plot the orientations of a single grain in one pole figure

I tried:
figure
plotPDF(ebsd_selected.orientations,m1,'antipodal')
hold on
plotPDF(ebsd_selected.orientations,m2,'antipodal')
hold on
plotPDF(ebsd_selected.orientations,m3,'antipodal')
hold off
where m1,m2 and m3 are Miller indices for 100,010 and 001 respectively, but the result of it was that it plots the orientations only according to m1.

can it be done?
Thanks,
Amir

**Comment by kilir** [comment, marked as answer]
Hi,
how about:
```
v = ebsd_selected.orientations*symmetrise([m1 m2 m3],'unique','antipodal');
plot(v,'antipodal')
```
Cheers,
Rüdiger

**Reply by da-levy** [reply]
I am trying to do the above, which worked, but I need to change the color key.
How do you color points in the plot() pole figure based on mis2mean angle?

When using plotPDF, I code: 
cs = ebsd('Apatite').CS;
h = Miller(0, 0, 1,cs);
plotPDF(ori, ebsd.mis2mean.angle/degree, h)

'ebsd.mis2mean.angle/degree' colors each point based on its mis2mean angle.

Can I do this for somehow for plot()?

Thanks!

**Reply by kilir** [reply]
Yes, with plot() it works exactly the same way as in plotPDF, 2nd argument is the data.
Cheers, Rüdiger

**Reply by da-levy** [reply]
Hi @kilir ,

I actually am not able to get this to work. When I run the below code, I get the following error: Warning: 
_Error creating or updating Patch
Error in value of property  FaceVertexCData
number of colors  must be equal to number of vertices when MarkerFaceColor is Flat_ 

This level of Matlab know-how is above my head, but it seems the dimensions of 'ebsd_subset.mis2mean.angle/degree' is incompatible with that of 'v'.
I am not sure how to correct this.

Attached are the figures from plotPDF() and plot() (plot without 'ebsd_subset.mis2mean.angle/degree'). 
The goal is to get all planes on the same plot with the mis2mean coloring.

Any help you can offer is much appreciated!
Best,
Drew

%% display the orientations in pole figures

% the involved symmetries
cs = ebsd_subset('Apatite').CS;

h = Miller({0,0,1},cs);
k = Miller({1,0,0},cs);
l = Miller({1,1,0},cs);
m = Miller({0,1,1},cs);
n = Miller({1,1,1},cs);
yy = [h,k,l,m,n];

% scatter map of axis
figure
plotPDF(ori,ebsd_subset.mis2mean.angle/degree,yy,'points','all','lower','MarkerSize',1); %All points


%% All planes on one plot
figure
v = ebsd_subset.orientations*symmetrise(yy, 'unique', 'antipodal');
plot(v, ebsd_subset.mis2mean.angle/degree, 'antipodal', 'MarkerSize', 1); %All points


![plotPDF](https://user-images.githubusercontent.com/68292240/109438837-8cfda380-79e0-11eb-9c6d-1278e29fe2a0.jpg)
![plot](https://user-images.githubusercontent.com/68292240/109438836-8c650d00-79e0-11eb-8e8c-29d5e283311e.jpg)

**Comment by zmichels** [comment]
I think after you symmetrise, the number does not match the initial number. Perhaps you could try doing the symmetrise in the call for plot later instead?

**Reply by da-levy** [reply]
Thanks for the suggestion.
**v = ebsd_subset.orientations*symmetrise(yy, 'unique', 'antipodal');** and **m = ebsd_subset.mis2mean.angle/degree** are both the same size. The main difference I see is 'v' is 4472x38 vector3d and 'm' is 4472x1 double.

Based on the error, it seems 'm' needs to be a list of rgb triplets. I am really unsure, though.

Are you suggesting placing symmetrise in the plot call?
Like: plot(ebsd_subset.orientations, ebsd_subset.mis2mean.angle/degree, symmetrise(yy)...)

Thanks,
Drew

**Reply by kilir** [reply]
Yes, data need to have the same length as the points you want to plot. Example:
```
mtexdata titanium
cs = ebsd('t').CS;
[grains,ebsd.grainId,ebsd.mis2mean]=ebsd.calcGrains;
% length ebsd('t')

h = [Miller(1,1,-2,0,cs) Miller(1,0,-1,1,cs)]
h = h.symmetrise('unique', 'antipodal')
% length of h is 9

v= ebsd('t').orientations*h;
%length of v is 8100 x 9

% hence we need to repeat mismean 9 times
m2m = repmat(ebsd('t').mis2mean,1,length(h))

plot(v,m2m.angle/degree)
```

Hope that helps.

Cheers,
Rüdiger
